test/specs/parsing/axes/DescendantAxis.tests.ts

```diff
@@ -14,44 +12,69 @@ beforeEach(() => {
 
 describe('descendant', () => {
 	it('parses descendant::', () => {
-		jsonMlMapper.parse([
-			'someParentElement',
-			['someElement']
-		], documentNode);
-		chai.assert.deepEqual(evaluateXPathToNodes('descendant::someElement', documentNode), [documentNode.firstChild.firstChild]);
+		jsonMlMapper.parse(['someParentElement', ['someElement']], documentNode);
+		chai.assert.deepEqual(evaluateXPathToNodes('descendant::someElement', documentNode), [
+			documentNode.firstChild.firstChild
+		]);
+	});
+
+	it('passes buckets for descendant', () => {
+		jsonMlMapper.parse(['parentElement', ['childElement']], documentNode);
+
+		const childNode = documentNode.firstChild.firstChild;
+
+		const testDomFacade: IDomFacade = {
+			getFirstChild: (node: slimdom.Node, bucket: string|null) => {
+				chai.assert.isNotNull(bucket, 'It has bucket');
```

Comment:
We can make this test whether it passes the bucket for 'childElement' (`getBucketsForSelector('self::childElement')`)